torchnlp/metrics/rouge.py

```diff
@@ -0,0 +1,332 @@
+                 beta=None, strict=False):
+    """
+    Computes the LCS-based F-measure score.
+    Args:
+       wlcs: wlcs score
+       m: number of words in reference summary
+       n: number of words in candidate summary
+       f: weighting function
+       inv_f: inverse function of weighting function
+       beta: beta = P_lcs / R_lcs when ∂ F_lcs / ∂ R_lcs = ∂ F_lcs / ∂ P_lcs.
+           In DUC, beta is set to a very big number, and only R_lcs is considered.
+    Returns:
+       dictionary. WLCS-based F-measure score, P-score and R-score
+    """
+    if strict:
+        assert(check_increase(f))
+        assert(check_inverse(f, inv_f))
+    r_wlcs = inv_f(wlcs / f(m))
+    p_wlcs = inv_f(wlcs / f(n))
+    if not beta:
+        beta = p_wlcs / (r_wlcs + 1e-12)
+    num = (1 + (beta**2)) * r_wlcs * p_wlcs
+    denom = r_wlcs + ((beta**2) * p_wlcs)
+    f_wlcs = num / (denom + 1e-12)
+    return {"f": f_wlcs, "p": p_wlcs, "r": r_wlcs}
+
+
+def lcs(x, y):
+    """
+    Compute the length of longest common sequence of x and y
+    Args:
+        x, y: List of element
+    Returns:
+        Dictionary, table[i, j] represent the length of
+            the longest common sequence of x[0: i-1] and y[0: j-1]
+    """
+    n, m = len(x), len(y)
+    table = {}
+    for i in range(n + 1):
+        for j in range(m + 1):
+            if i == 0 or j == 0:
+                table[i, j] = 0
+            elif x[i - 1] == y[j - 1]:
+                table[i, j] = table[i - 1, j - 1] + 1
+            else:
+                table[i, j] = max(table[i - 1, j], table[i, j - 1])
+    return table
+
+
+def lcs_seq(x, y):
+    """
+    Find the longest common sequence of x and y
+    Args:
+        x, y: List of element
+    Returns:
+        List, the longest common sequence of x and y
+    """
+    table = lcs(x, y)
+
+    def lcs_seq_wrd(i, j):
+        if i == 0 or j == 0:
+            return []
+        if x[i - 1] == y[j - 1]:
+            return lcs_seq_wrd(i - 1, j - 1) + [x[i - 1]]
+        elif table[i - 1, j] > table[i, j - 1]:
+            return lcs_seq_wrd(i - 1, j)
+        else:
+            return lcs_seq_wrd(i, j - 1)
+    return lcs_seq_wrd(len(x), len(y))
+
+
+def _w_lcs(x, y, func=lambda x: x**2):
+    """
+    Compute the we of weighted longest common sequence of x and y,
+
+    c is the dynamic programming table, c(i,j) stores
+        the WLCS score ending at word x[i] of X and y[j] of Y.
+    w is the table storing the length of consecutive matches ended at c table position i and j,
+        and f is a function of consecutive matches at the table position, c(i, j).
+
+    Args:
+        x, y: List of element
+        func: the weighting function which should satisfies f(x+y) > f(x) + f(y) for
+            any positive integers x and y, and should hava a close form inverse function.
+    Returns:
+        Float, the WLCS score of x and y
+    """
+    n = len(x)
+    m = len(y)
```

Comment:
Can you please use full variable names? It's difficult to read code that uses single variable names?